### PR TITLE
refactor esdb nack actions

### LINKED ISSUE #727: RFC: Unmarshall errors should not be retried
### Describe the chore

We have numerous snippets like this:
```
if err := evt.GetJsonData(&payload); err != nil {
		return err, esdb.NackActionRetry
	}
```

But if an event can't be processed because it is invalid (syntactically, or schema-wise), it will not be processed the next time around. Only after a significant time has passed, and a developer pushed a change, the event might be handled.
So if a user changes the description of a task and something goes wrong in the Command handler, yielding a bad event, not only is there no error emitted, but the description might change weeks later, possibly overwriting changes made since.

### PR DESCRIPTION
**Pull request by PaulKalho**
closes #727

### CODE REVIEW

**Comment by PaulKalho** [comment]
`NackActionRetry` events will be retried 10 times. We could change this to any value in  `CreatePersistentSubscriptionToAll()`

**Comment by PaulKalho** [comment]
@MaxSchaefer I observed that `NackActionUnknown` will be treated like `NackActionRetry` atm. I am not quite sure where/if we can change that behavior.

File: services/property-svc/internal/property/projections/property_postgres_projection/property_postgres_projection.go [hunk 1]
```diff
@@ -417,7 +417,7 @@ func (p *Projection) onFieldTypeDataSelectOptionsUpserted(ctx context.Context, e
 		} else {
 			if selOpt.Name == nil {
 				log.Error().Msg("selectOption name has to be set on create")
-				return nil, esdb.NackActionRetry
+				return nil, hwutil.PtrTo(esdb.NackActionRetry)
```
Comment (MaxSchaefer) [inline services/property-svc/internal/property/projections/property_postgres_projection/property_postgres_projection.go:420]:
```suggestion
				// existsSelectOptions gets modified by a database result. Thats why we are retrying here.
				return nil, hwutil.PtrTo(esdb.NackActionRetry)
```

**Review by MaxSchaefer — APPROVED** [review]
`PtrTo` `PtrTo` `PtrTo` `PtrTo` `PtrTo` `PtrTo` we can probably introduce some sort of own enum later on that maps to esdb 👍

**Pull request merged**